### ISSUE TITLE
Updated Intel drivers break GPU-PV?

### ISSUE THREAD

**Issue by ngorevsk** [jamesstringerparsec/Easy-GPU-PV#133]
My host is Windows 11 and guests are Windows 10. My GPU-PV had been working without issue until my host had its video card drivers updated from Windows Update. My "Intel(R) UHD Graphics 730" video drivers were updated on the host and caused GPU-PV to stop working on the guest VMs - understandably due to the mismatch (host = new drivers, guest = old drivers). I would get a code 43 error in device manager and matching the driver folder on the guest would not resolve it.

Rolling back host drivers was the only fix I could come up with. I used device manager to update my host video card driver to something I knew previously worked from a few months back (7/23/2021 v30.0.100.9805) and matched the driver folder on the guest and this fixed my issue.

I really can't make sense of this. To me this seems to suggest that GPU-PV may not work with newer Intel drivers. Has anyone else seen this?

My half-solution for now is to use gpedit.msc and enable a setting "do not include drivers with windows update policy" to ensure the drivers don't automatically change again without my explicit action.

**Comment by WanpengQian** [comment]
From what I know, Guest's GPU driver should always sync to host's.
If host's driver update, you have to manual update guest's driver too.

You can have a try:
https://gist.github.com/neg2led/e35793da476095beac716c16ffba1d23
Download [New-GPUPDriverPackage.ps1](https://gist.github.com/neg2led/e35793da476095beac716c16ffba1d23#file-new-gpupdriverpackage-ps1) and execute as administrator within PowerShell. it will collect the host's GPU driver files.
then transfer the zip file to guest. unzip it. copy/move them to C:\Windows folder. 
Reboot guest. it should works.

**Comment by ngorevsk** [comment]
I agree that it should work and I appreciate you trying to help by linking to a guide but that is what I did (manual steps but for my PCs Intel drivers instead). It has worked for me before. Unfortunately matching the latest Intel drivers on both host and guest is currently not working for me so I am forced to stick with the older drivers to get GPU-PV working on my guests.

I'm curious if anyone else has seen this happen, to better gauge if this is a wider problem or "just me."

I suppose it might help if I mentioned my OS versions. 
Host: Windows 11 Pro 21H2
Guests: Windows 10 Pro 20H2, Windows 10 Pro 21H1, and Windows 11 Pro 21H2.

**Comment by ngorevsk** [comment]
I gave this another shot since I wanted the VM host to use Parsec but that required the latest Intel drivers as my current ones were too old.

Using that "New-GPUPDriverPackage.ps1" script and subsequent copying of drivers unfortunately didn't work for my existing VMs - they were still getting (code 43) error on the video card in device manager.

Interestingly enough, the script and subsequent copying of the drivers to a new Win 11 VM worked perfectly! The host and guest are both working as expected with the latest Intel drivers.

I wonder if copying the multiple driver versions to the guest VM somehow confuses the GPU-PV-guest-driver-detection resulting in that code 43 error? I can revisit this theory the next time I upgrade drivers...

Using that script was definitely an improvement to me doing things manually - so thanks for the tip @WanpengQian!
